internal/controller/api_to_config.go

```diff
@@ -67,8 +67,37 @@ func apiToFRR(resources ClusterResources, alwaysBlock []net.IPNet) (*frr.Config,
 		}
 
 		alwaysBlockFRR := alwaysBlockToFRR(alwaysBlock)
+		prefixesForVRF := map[string][]string{}
 		for _, r := range cfg.Spec.BGP.Routers {
-			routerCfg, err := routerToFRRConfig(r, alwaysBlockFRR, resources.PasswordSecrets, bfdProfiles)
+			err := validatePrefixes(r.Prefixes)
+			if err != nil {
+				return nil, err
+			}
+			prefixesForVRF[r.VRF] = r.Prefixes
+		}
+
+		for _, r := range cfg.Spec.BGP.Routers {
+			err := validateImportVRFs(r, prefixesForVRF)
+			if err != nil {
+				return nil, err
+			}
+
+			// All prefixes available in this router: defined and imported from VRFs
+			prefixesInRouter := sets.New(r.Prefixes...)
+			for _, i := range r.Imports {
+				importedPrefixes := prefixesForVRF[i.VRF]
```

Comment:
the problem here is, frr wants "default" when importing, while in the api is cleaner not to bother about "default" VRF if you don't have to deal with VRFs at all. Because of that, I couldn't find a better way to do this translation. I think having a slimmer api has more value over this.

Re - key not existing I need to recheck.

Reply:
I'll find a way to make it clearer. The missing key can't happen because we validate that imported vrfs exist and we fill a record for each vrf, but I agree it's weak and we should at least log (or even better panic) if the key is not there as we think it cna't happen.